cli/dcoscli/tables.py

```diff
@@ -849,10 +850,31 @@ def slave_table(slaves):
         ('ID', lambda s: s['id'])
     ])
 
-    tb = table(fields, slaves, sortby="HOSTNAME")
+    for field_name in field_names:
+        if field_name.lower() == 'ip':
+            fields[field_name.upper()] = lambda s: mesos.parse_pid(s['pid'])[1]
+        else:
+            if ':' in field_name:
+                heading, field_name = field_name.split(':', 1)
+            else:
+                heading = field_name
+            fields[heading.upper()] = _dotted_itemgetter(field_name.lower())
+
+    sortby = list(fields.keys())[0]
+    kwargs = {}
+    if sortby.lower() == 'ip':
+        kwargs['sort_key'] = lambda d: tuple(int(x) for x in d[0].split('.'))
+    tb = table(fields, slaves, sortby=sortby, **kwargs)
```

Comment:
Yeah and that's actually what happens, because we always set `sortby` to the first field:
```python
sortby = list(fields.keys())[0]
```

and now the first field is always `HOSTNAME`.

```
$ dcos node --field=ip --field='disk used:used_resources.disk'
HOSTNAME       IP                          ID                    DISK USED
   A      10.10.12.29   74df965e-a46e-46a9-8d2a-639db0af0e25-S2    5096.0
   B      10.10.12.174  74df965e-a46e-46a9-8d2a-639db0af0e25-S0    4096.0
   C      10.10.12.183  74df965e-a46e-46a9-8d2a-639db0af0e25-S4    4096.0
   D      10.10.3.201   74df965e-a46e-46a9-8d2a-639db0af0e25-S3     0.0
```

The code in lines 865-866 is only to do numeric sorting instead of alphabetic sorting if the `sortby` field happens to be `ip`. Which I guess can't really happen now, because `hostname` will always be the first field, but maybe in the future, we might allow switching up the order or changing the sort field, so maybe I should leave it?